pkg/operator/k8sutil/service.go

```diff
@@ -92,3 +100,103 @@ func ParseServiceType(serviceString string) v1.ServiceType {
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, errors.Wrapf(err, "failed to get exported service %q", service.Name)
+	}
+
+	return true, nil
+}
+
+// ExportService exports the service using MCS API and returns the external IP of the exported service
+func ExportService(ctx context.Context, c *clusterd.Context, service *v1.Service) (string, error) {
+	client, err := mcsv1Client.NewForConfig(c.KubeConfig)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get mcs-api client")
+	}
+
+	serviceExport := &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            service.Name,
+			Namespace:       service.Namespace,
+			OwnerReferences: service.GetOwnerReferences(),
+		},
+	}
+
+	_, err = client.MulticlusterV1alpha1().ServiceExports(service.Namespace).Create(ctx, serviceExport, metav1.CreateOptions{})
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return "", errors.Wrapf(err, "failed to create exported service %q", service.Name)
+	}
+
+	var exportedIP string
+	var serviceExportError error
+	exportedIP, err = getExportedServiceIP(fmt.Sprintf("%s.%s.svc.clusterset.local", service.Name, service.Namespace))
+
+	if err != nil {
+		serviceExportError = errors.Wrapf(err, "failed to get exported service IP for %q", service.Name)
+		err := verifyExportedService(ctx, client, service.Name, service.Namespace)
+		if err != nil {
+			serviceExportError = errors.Wrapf(err, "failed to create service export successfully for the service %q", service.Name)
+		}
+		return "", serviceExportError
+	}
+
+	return exportedIP, nil
+}
+
+// verifies if the ServiceExport status conditions to determine if the service was exported correctly
+func verifyExportedService(ctx context.Context, client *mcsv1Client.Clientset, name, namespace string) error {
+	exportedService, err := client.MulticlusterV1alpha1().ServiceExports(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get exported service %q", name)
+	}
+
+	if len(exportedService.Status.Conditions) == 0 {
+		return fmt.Errorf("no status conditions available in the exported service %q", name)
+	}
+
+	for _, condition := range exportedService.Status.Conditions {
+		if condition.Type == mcsv1a1.ServiceExportValid && condition.Status == corev1.ConditionFalse {
+			return fmt.Errorf(*condition.Message)
+		}
+	}
+
+	return nil
+}
+
+func getExportedServiceIP(fqdn string) (string, error) {
+	retryCount := 5
+	serviceExportWaitTime := 10 * time.Second
+
+	ips := []net.IP{}
+	var err error
+	for i := 0; i < retryCount; i++ {
+		ips, err = net.LookupIP(fqdn)
+		if err != nil {
+			if i < retryCount-1 {
+				logger.Warningf("failed to resolve DNS for %q. Trying again...", fqdn)
+				time.Sleep(serviceExportWaitTime)
+				continue
+			} else {
+				return "", errors.Wrapf(err, "failed to resolve DNS for %q. %v", fqdn, err)
+			}
+		}
+		logger.Debugf("available addresses for %q : %v", fqdn, ips)
```

Comment:
Don't we need to return here since it was successful? Otherwise, looks like we will continue looping.
```suggestion
		logger.Debugf("available addresses for %q : %v", fqdn, ips)
		return ips[0].String(), nil
```